sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -86,13 +87,50 @@ case class DescribeDatabase(
     extended: Boolean)(sql: String)
   extends NativeDDLCommand(sql) with Logging
 
+/**
+ * The DDL command that creates a function.
+ * alias: the class name that implements the created function.
+ * resources: Jars, files, or archives which need to be added to the environment when the function
+ *            is referenced for the first time by a session.
+ * isTemp: indicates if it is a temporary function.
+ */
 case class CreateFunction(
     databaseName: Option[String],
     functionName: String,
     alias: String,
     resources: Seq[(String, String)],
     isTemp: Boolean)(sql: String)
-  extends NativeDDLCommand(sql) with Logging
+  extends NativeDDLCommand(sql) with Logging {
+
+  override def run(sqlContext: SQLContext): Seq[Row] = {
+    val func = FunctionIdentifier(functionName, databaseName)
+    val catalogFunc = CatalogFunction(func, alias)
+    if (isTemp) {
+      // Set `ignoreIfExists` to false, so if the temporary function already exists,
+      // an exception will be thrown.
+      val (info, builder) =
+        sqlContext.sessionState.functionRegistry.getFunctionBuilderAndInfo(functionName, alias)
+      sqlContext.sessionState.catalog.createTempFunction(functionName, info, builder, false)
```

Comment:
So rename it to `genFunctionBuilderAndInfo`? Actually it is going to generate a function builder and expression info for us in order to create temporary function. But I think it is just the naming problem to make you confusing.